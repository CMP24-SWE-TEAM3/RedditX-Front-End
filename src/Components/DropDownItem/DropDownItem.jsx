import React from 'react';
import Dropdown from "react-bootstrap/Dropdown";
import {DropdownContainer} from "./DropDownItem.styled";
import CustomMenu from "Components/CustomMenu/CustomMenu";
import db from "Data/db.json"

/**
 * Component that displays a dropdown item in left-hand side corner of the home page
 * @return {React.Component}
 */
const DropDownItem = () => {
    return (
        <DropdownContainer as={CustomMenu}>
            <Dropdown.ItemText as={"title"}>your communities</Dropdown.ItemText>
            {db.Communities.map(community => {
                return (
                    <Dropdown.Item key={community.id} eventKey={community.id}>{`r/${community.name}`}</Dropdown.Item>
                )
            })}
            <Dropdown.ItemText as={'title'}>FEEDS</Dropdown.ItemText>
            {db.Feeds.map(feed => {
                return (
                    <Dropdown.Item key={feed.id} eventKey={feed.id}>{feed.name}</Dropdown.Item>
                )
            })}

            <Dropdown.ItemText as={'title'}>OTHERS</Dropdown.ItemText>
            {db.OTHERS.map(other => {
                return (
                    <Dropdown.Item key={other.id} eventKey={other.id}>{other.name}</Dropdown.Item>
                )
            })}
        </DropdownContainer>
    );
};
export default DropDownItem;

