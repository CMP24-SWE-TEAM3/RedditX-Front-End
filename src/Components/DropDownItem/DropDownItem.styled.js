import styled from "styled-components";

export const DropdownContainer = styled.div`
  .ico {
    height: 20px;
    width: 20px;
    border-radius: 100%;
    vertical-align: middle;
  }


  .community {
    //padding: 8px 24px;
    position: relative;
    align-items: center;
    display: flex;
    flex-direction: row;
  }

  .word {
    flex-grow: 1;
    margin-left: 8px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }


  span {
    flex-grow: 0;
    height: 20px;
    max-height: 20px;
    margin-right: 5px;
    max-width: 20px;
    width: 20px;
  }

  .right-span {
    margin-left: 8px;
  }

  .feed {
    padding: 16px 24px 8px;
    //color: #878a8c;
    color: ${({theme}) => theme.color.primary};
    font-size: 10px;
    font-weight: 500;
    line-height: 16px;
    text-transform: uppercase;
  }

  .dropdown-item {
    color: ${({theme}) => theme.color.primary} !important;
    padding: 8px 24px;
    position: relative;
    text-decoration: none;
    flex-grow: 1;
    font-size: 14px;
    line-height: 18px;
    margin-left: 8px;
    max-width: 208px;
    overflow: hidden;
    text-align: left;
    text-overflow: ellipsis;
    white-space: nowrap;

    :active {
      background-color: unset !important;
    }
    :hover{
      background-color:${({theme}) => theme.lineColor.primary};
    }
  }

  .form-control {
    background-color: ${({theme}) => theme.background.post_background};
    border: 1px solid ${({theme}) => theme.color.secondary};
    color: ${({theme}) => theme.color.primary};
    height: 30px;
    margin: 8px 16px 16px ;
    outline: none;
    padding: 0 6px;
    font-size: 14px;
    font-weight: 400;
    line-height: 21px;
    font-family: inherit;
    //align-items:center;
    box-sizing: border-box;
    border-radius: 0;
    width: calc(100% - 32px);
  }

  .form-control:focus {
    border-color: ${({theme}) => theme.color.secondary} !important;
    box-shadow: none;
  }
  .dropdown-item-text{
    font-size: 10px;
    font-weight: 500;
    line-height: 16px;
    text-transform: uppercase;
    padding: 16px 24px 8px;
    color: #878a8c;
  }
`