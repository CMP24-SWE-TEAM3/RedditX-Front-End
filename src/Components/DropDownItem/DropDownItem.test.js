import {shallow} from 'enzyme';

// import Components
import DropDownItem from "./DropDownItem";

describe("Drop Down Item Component", () => {
    it("this is a test for Drop Down Item Component", () => {
        expect(shallow(<DropDownItem/>)).toMatchSnapshot();

    });
});