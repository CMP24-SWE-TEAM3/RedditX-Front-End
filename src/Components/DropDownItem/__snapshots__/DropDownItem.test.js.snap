// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Down Item Component this is a test for Drop Down Item Component 1`] = `ShallowWrapper {}`;
